pandas/core/indexes/category.py

```diff
@@ -198,8 +200,14 @@ def __new__(
                 data = []
 
         assert isinstance(dtype, CategoricalDtype), dtype
-        if not isinstance(data, Categorical) or data.dtype != dtype:
+        if isinstance(data, (cls, ABCSeries)) and is_categorical_dtype(data):
+            data = extract_array(data)
```

Comment:
extract_array makes the `isinstance` check above unnecessary, also pls pass `extract_numpy=True`